### ISSUE TITLE
decryption of arclink download: Unicode error

### ISSUE THREAD

**Issue by miic-sw** [obspy/obspy#1347]
I get a UnicodeDecodeError when downloading encrypted data with the arclink client. It occurs when the key and the iv are created.

I use 
ObsPy 1.0.0
Python 2.7.11
Anaconda 2.4.1
Ubuntu 14.04

The M2Crypto package I have installed is
m2crypto 0.23.0  py27_1  anaconda-cluster

Here is the code I used to download

```python
from obspy import UTCDateTime
from obspy.clients.arclink.client import Client
client = Client(user='sens-schoenfelder@gfz-potsdam.de',debug=True)
t = UTCDateTime(2015, 6, 17, 0, 0,0)  
st = client.get_waveforms("XN", "NEP05", "*", "HHZ", t,t+2,compressed=False)
```

To check the delivery form the datacenter I commented out lines 375 and 376 form arclink/client.py `data = decryptor.update(data); data += decryptor.final()`

and saved the un-decrypted stream in file_stream.openssl after  execution of 

```python
self.save_waveforms(file_stream, network, station, location, channel,
                            starttime, endtime, format=format,
                            compressed=compressed, route=route)
```

on line 444.

This can be decrypted correctly with openssl
```
openssl des-cbc -pass pass:'my_password' -in file_stream.openssl -out stream.mseed -d
```

The UnicodeError does not depend on my password. But if anyone looks into this I can give you the password.

here are the files:
encrypted: https://www.dropbox.com/s/dijvkb3wgg5z3ie/stream.mseed.openssl?dl=0
decrypted: https://www.dropbox.com/s/go7v3p5aye7wz3o/stream.mseed?dl=0

And finally here is the full output:

```
Connected to webdc.eu:18002
>>> HELLO

... ArcLink v1.2 (2013.070)
... GFZ
>>> USER sens-schoenfelder@gfz-potsdam.de

... OK
>>> INSTITUTION Anonymous

... OK
>>> REQUEST ROUTING 

>>> 2015,6,16,23,59,59,0 2015,6,17,0,0,3,0 XN NEP05

>>> END

... OK
... 34994076
>>> STATUS 34994076

... <?xml version="1.0"?>
<arclink>
  <request id="34994076" type="ROUTING" label="" args="" encrypted="false" size="0" ready="false" error="false" message="">
    <volume id="UNSET" status="UNSET" size="0" message="">
      <line content="2015,6,16,23,59,59,0 2015,6,17,0,0,3,0 XN NEP05" status="UNSET" size="0" message=""/>
    </volume>
  </request>
</arclink>
END
>>> STATUS 34994076

... <?xml version="1.0"?>
<arclink>
  <request id="34994076" type="ROUTING" label="" args="" encrypted="false" size="390" ready="true" error="false" message="">
    <volume id="routing" dcid="GFZ" status="OK" size="390" encrypted="false" message="">
      <line content="2015,6,16,23,59,59,0 2015,6,17,0,0,3,0 XN NEP05" status="OK" size="0" message=""/>
    </volume>
  </request>
</arclink>
END
>>> DOWNLOAD 34994076

<?xml version="1.0" encoding="utf-8"?>
<ns0:routing xmlns:ns0="http://geofon.gfz-potsdam.de/ns/Routing/1.0/">
	<ns0:route locationCode="" networkCode="XN" publicID="Route#20150305195044.937296.9213" stationCode="" streamCode="">
		<ns0:arclink address="eida.gfz-potsdam.de:18002" end="2020-12-31T00:00:00.0000Z" priority="1" start="2015-01-01T00:00:00.0000Z" />
	</ns0:route>
</ns0:routing>
>>> PURGE 34994076

>>> BYE


Requesting eida.gfz-potsdam.de:18002
>>> HELLO

... ArcLink v1.2 (2013.070)
... GFZ
>>> USER sens-schoenfelder@gfz-potsdam.de

... OK
>>> INSTITUTION Anonymous

... OK
>>> REQUEST WAVEFORM format=MSEED

>>> 2015,6,16,23,59,59,0 2015,6,17,0,0,3,0 XN NEP05 HHZ *

>>> END

... OK
... 34994078
>>> STATUS 34994078

... <?xml version="1.0"?>
<arclink>
  <request id="34994078" type="WAVEFORM" label="" args="format=MSEED" encrypted="false" size="0" ready="false" error="false" message="">
    <volume id="UNSET" status="UNSET" size="0" message="">
      <line content="2015,6,16,23,59,59,0 2015,6,17,0,0,3,0 XN NEP05 HHZ *" status="UNSET" size="0" message=""/>
    </volume>
  </request>
</arclink>
END
>>> STATUS 34994078

... <?xml version="1.0"?>
<arclink>
  <request id="34994078" type="WAVEFORM" label="" args="format=MSEED" encrypted="true" size="24" ready="false" error="false" message="">
    <volume id="UNSET" status="UNSET" size="0" message="">
      <line content="2015,6,16,23,59,59,0 2015,6,17,0,0,3,0 XN NEP05 HHZ *" status="UNSET" size="0" message=""/>
    </volume>
  </request>
</arclink>
END
>>> STATUS 34994078

... <?xml version="1.0"?>
<arclink>
  <request id="34994078" type="WAVEFORM" label="" args="format=MSEED" encrypted="true" size="37527576" ready="false" error="false" message="">
    <volume id="GFZ" dcid="GFZ" status="OK" size="37527576" encrypted="true" message="">
      <line content="2015,6,16,23,59,59,0 2015,6,17,0,0,3,0 XN NEP05 HHZ *" status="OK" size="37527552" message=""/>
    </volume>
  </request>
</arclink>
END
>>> STATUS 34994078

... <?xml version="1.0"?>
<arclink>
  <request id="34994078" type="WAVEFORM" label="" args="format=MSEED" encrypted="true" size="37527576" ready="true" error="false" message="">
    <volume id="GFZ" dcid="GFZ" status="OK" size="37527576" encrypted="true" message="">
      <line content="2015,6,16,23,59,59,0 2015,6,17,0,0,3,0 XN NEP05 HHZ *" status="OK" size="37527552" message=""/>
    </volume>
  </request>
</arclink>
END
>>> DOWNLOAD 34994078

37527576 bytes of data read
>>> PURGE 34994078

>>> BYE

Traceback (most recent call last):
  File "encrypt_check.py", line 7, in <module>
    st = client.get_waveforms("XN", "NEP05", "*", "HHZ", t,t+2,compressed=False)
  File "/home/chris/lib/anaconda2/lib/python2.7/site-packages/obspy/clients/arclink/client.py", line 442, in get_waveforms
    compressed=compressed, route=route)
  File "/home/chris/lib/anaconda2/lib/python2.7/site-packages/obspy/clients/arclink/client.py", line 566, in save_waveforms
    data = self._fetch(rtype, rdata, route=route)
  File "/home/chris/lib/anaconda2/lib/python2.7/site-packages/obspy/clients/arclink/client.py", line 261, in _fetch
    return self._request(request_type, request_data)
  File "/home/chris/lib/anaconda2/lib/python2.7/site-packages/obspy/clients/arclink/client.py", line 373, in _request
    data = decryptor.update(data)
  File "/home/chris/lib/anaconda2/lib/python2.7/site-packages/obspy/clients/arclink/decrypt.py", line 43, in update
    [key, iv] = self._get_key_iv(self._password, chunk[8:16])
  File "/home/chris/lib/anaconda2/lib/python2.7/site-packages/obspy/clients/arclink/decrypt.py", line 71, in _get_key_iv
    key += chunk[0:i]
UnicodeDecodeError: 'ascii' codec can't decode byte 0x97 in position 3: ordinal not in range(128)
```

**Comment by barsch** [comment]
related to #1314

**Comment by barsch** [comment]
Please send me your pw to barsch@egu.eu - I'll take a look this evening

**Comment by miic-sw** [comment]
I read the #1314 and I don't think it is related. The password properly arrives in _get_key_iv as a unicode string. u'my_password'

**Comment by barsch** [comment]
could be - but using your credentials I am maybe able to solve this too - will see ...

**Comment by miic-sw** [comment]
Thanks! Works fine.

**Comment by barsch** [comment]
no problem - but not done yet - cant get M2Crypto to work for Py3k ... currently looking into pycrypto as an alternative :/

**Comment by megies** [comment]
Closed by 67603d3d5ded2f864a8a310f88f.
